src/packet/udp.rs

```diff
@@ -101,6 +50,8 @@ fn udp_header_ipv4_test() {
     packet[20 + 8 + 3] = 't' as u8;
 
     {
+        use pnet::packet::ipv4::Ipv4Packet;
+        let ip_header = Ipv4Packet::new(&packet[..]).unwrap();
         let mut udp_header = MutableUdpPacket::new(&mut packet[20..]).unwrap();
```

Comment:
I should note that the other errors you're getting are caused for the same reason. Since this is just a unit test, you could cheat and use http://doc.rust-lang.org/std/primitive.slice.html#method.split_mut - ultimately I think there needs to be a better solution though.